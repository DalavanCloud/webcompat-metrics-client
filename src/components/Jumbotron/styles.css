/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

:root {
  --fontSize: var(--fontSize);
}

.component {
  font-size: var(--fontSize);
  padding: 1.5em;
  border-bottom: 1px solid #dee5e7;
  color: #58666e;
  margin: 0 0 2em 0;
}

.title {
  margin: 0;
  font-weight: 300;
}

.subtitle {
  font-size: 1em;
  margin: 0;
  font-weight: 300;
}
