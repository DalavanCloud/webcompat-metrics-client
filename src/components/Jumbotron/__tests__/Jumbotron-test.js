/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import React from "react";
import renderer from "react-test-renderer";

import Jumbotron from "../";

it("renders Jumbotron default correctly with a title", () => {
  const tree = renderer.create(<Jumbotron title="Jumbotron title" />).toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Jumbotron default correctly with a title and a subtitle", () => {
  const tree = renderer
    .create(<Jumbotron title="Jumbotron title" subtitle="Jumbotron subtitle" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it("renders Jumbotron default correctly with a style props", () => {
  const tree = renderer
    .create(
      <Jumbotron
        style={{ fontSize: "2rem" }}
        title="Jumbotron title"
        subtitle="Jumbotron subtitle"
      />,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
