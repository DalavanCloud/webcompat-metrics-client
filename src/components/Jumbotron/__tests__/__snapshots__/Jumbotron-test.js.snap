// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Jumbotron default correctly with a style props 1`] = `
<div
  className="component"
  style={
    Object {
      "fontSize": "2rem",
    }
  }
>
  <h2
    className="title"
  >
    Jumbotron title
  </h2>
  <h3
    className="subtitle"
  >
    Jumbotron subtitle
  </h3>
</div>
`;

exports[`renders Jumbotron default correctly with a title 1`] = `
<div
  className="component"
  style={Object {}}
>
  <h2
    className="title"
  >
    Jumbotron title
  </h2>
</div>
`;

exports[`renders Jumbotron default correctly with a title and a subtitle 1`] = `
<div
  className="component"
  style={Object {}}
>
  <h2
    className="title"
  >
    Jumbotron title
  </h2>
  <h3
    className="subtitle"
  >
    Jumbotron subtitle
  </h3>
</div>
`;
